polyfill/lib/calendar.mjs

```diff
@@ -45,126 +45,162 @@ export class Calendar {
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].dayOfYear(date);
   }
   weekOfYear(date) {
-    void date;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].weekOfYear(date);
   }
   daysInWeek(date) {
-    void date;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].daysInWeek(date);
   }
   daysInMonth(date) {
-    void date;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].daysInMonth(date);
   }
   daysInYear(date) {
-    void date;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].daysInYear(date);
   }
   monthsInYear(date) {
-    void date;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].monthsInYear(date);
   }
   inLeapYear(date) {
-    void date;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].inLeapYear(date);
   }
   hour(time) {
-    void time;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].hour(time);
   }
   minute(time) {
-    void time;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].minute(time);
   }
   second(time) {
-    void time;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].second(time);
   }
   millisecond(time) {
-    void time;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].millisecond(time);
   }
   microsecond(time) {
-    void time;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].microsecond(time);
   }
   nanosecond(time) {
-    void time;
-    throw new Error('not implemented');
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return impl[GetSlot(this, CALENDAR_ID)].nanosecond(time);
   }
   toString() {
     if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
```

Comment:
SGTM as long as docs for custom time zones and custom calendars are updated.  

> Worst case for serialization is that the correct ISO date (/...) is returned with an incorrect `[c=..]` annotation, right? That doesn't seem all that terrible.

It would prevent deserializing into your custom calendar but probably wouldn't crash anything, so might be an easy bug to slip into production.  But this is an advanced use case so I'm not worried much either way as long as it's documented clearly what the developer should do.

Reply:
I've updated the docs (which I had forgotten, and needed to be updated anyway)